tests/src/test/java/io/streamnative/pulsar/handlers/kop/InnerTopicProtectionTest.java

```diff
@@ -0,0 +1,185 @@
+import org.apache.pulsar.client.admin.PulsarAdminException;
+import org.apache.pulsar.common.policies.data.ClusterData;
+import org.apache.pulsar.common.policies.data.RetentionPolicies;
+import org.apache.pulsar.common.policies.data.TenantInfo;
+import org.testng.Assert;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+/**
+ * Inner topic protection test.
+ */
+@Slf4j
+public class InnerTopicProtectionTest extends KopProtocolHandlerTestBase {
+
+    protected int offsetsTopicNumPartitions;
+
+    protected KafkaServiceConfiguration resetConfig(int brokerPort, int webPort, int kafkaPort) {
+        KafkaServiceConfiguration kConfig = new KafkaServiceConfiguration();
+        kConfig.setBrokerServicePort(Optional.ofNullable(brokerPort));
+        kConfig.setWebServicePort(Optional.ofNullable(webPort));
+        kConfig.setListeners(PLAINTEXT_PREFIX + "localhost:" + kafkaPort);
+
+        kConfig.setOffsetsTopicNumPartitions(offsetsTopicNumPartitions);
+        kConfig.setEnableGroupCoordinator(true);
+
+        kConfig.setAdvertisedAddress("localhost");
+        kConfig.setClusterName(configClusterName);
+        kConfig.setManagedLedgerCacheSizeMB(8);
+        kConfig.setActiveConsumerFailoverDelayTimeMillis(0);
+        kConfig.setDefaultNumberOfNamespaceBundles(2);
+        kConfig.setZookeeperServers("localhost:2181");
+        kConfig.setConfigurationStoreServers("localhost:3181");
+        kConfig.setEnableGroupCoordinator(true);
+        kConfig.setAuthenticationEnabled(false);
+        kConfig.setAuthorizationEnabled(false);
+        kConfig.setAllowAutoTopicCreation(true);
+        kConfig.setAllowAutoTopicCreationType("partitioned");
+        kConfig.setBrokerDeleteInactiveTopicsEnabled(false);
+        //kConfig.setSystemTopicEnabled(true);
+        //kConfig.setTopicLevelPoliciesEnabled(true);
+
+        // set protocol related config
+        URL testHandlerUrl = this.getClass().getClassLoader().getResource("test-protocol-handler.nar");
+        Path handlerPath;
+        try {
+            handlerPath = Paths.get(testHandlerUrl.toURI());
+        } catch (Exception e) {
+            log.error("failed to get handler Path, handlerUrl: {}. Exception: ", testHandlerUrl, e);
+            return null;
+        }
+
+        String protocolHandlerDir = handlerPath.toFile().getParent();
+
+        kConfig.setProtocolHandlerDirectory(
+            protocolHandlerDir
+        );
+        kConfig.setMessagingProtocols(Sets.newHashSet("kafka"));
+
+        return kConfig;
+    }
+
+    @Override
+    protected void resetConfig() {
+        offsetsTopicNumPartitions = 16;
+        conf = resetConfig(
+            brokerPort,
+            brokerWebservicePort,
+            kafkaBrokerPort);
+
+        log.info("Ports --  broker: {}, brokerWeb:{}, kafka: {}",
+            brokerPort, brokerWebservicePort, kafkaBrokerPort);
+    }
```

Comment:
Could you explain why overwrite `resetConfig` here? I tried to remove the `resetConfig` code, and the tests still passed.

Reply:
The test can cover turn on system topic and topic policy with resetConfig. However, KOP can't work when turn on topic policy. In the end, it turn off the topic policy in resetConfig. We can turn on when it fixed.

Reply:
So I think these two `resetConfig` could be removed as well.

Reply:
OK